### PR TITLE
Feat/email templates

### PR DESCRIPTION
**Pull request by hcarneiro**
Form builder email templates and UI tweaks

### CODE REVIEW

File: js/libs/builder.js [hunk 1]
```diff
@@ -191,21 +196,127 @@ var app = new Vue({
...
     'permissionToChange': function(newVal) {
       Fliplet.Widget.toggleSaveButton(newVal);
     },
     'isAddingFields': function(newVal) {
       if (newVal) {
         Fliplet.Studio.emit('widget-mode', 'wide');
-        setTimeout(() => {
-          attatchObservers()
-        }, 1)
+        setTimeout(function() {
```
Comment (tonytlwu) [inline js/libs/builder.js]: Why the change from `() =>{}` to `function() {}`?

Comment (hcarneiro) [inline js/libs/builder.js]: must have copied from somewhere else!

File: js/libs/builder.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 var data = Fliplet.Widget.getData() || {};
 
 function changeSelectText() {
-  setTimeout(function() {
+  setTimeout(() => {
```
Comment (squallstar) [inline js/libs/builder.js]: don't use ES6, not all devices support it.

File: js/libs/builder.js [hunk 3]
```diff
@@ -83,7 +84,10 @@ var app = new Vue({
       toChangeTemplate: false,
       permissionToChange: false,
       newTemplate: '',
-      redirect: formSettings.redirect
+      redirect: formSettings.redirect,
+      toggleTemplatedEmail: formSettings.onSubmit.indexOf("templatedEmail") > -1 ? true : false,
```
Comment (squallstar) [inline js/libs/builder.js]: single quotes everywhere please :)
also: no need to do `? true : false`, just write the operation (`foo > -1`)

File: js/libs/builder.js [hunk 4]
```diff
@@ -170,8 +175,8 @@ var app = new Vue({
       }
 
       setTimeout(() => {
```
Comment (squallstar) [inline js/libs/builder.js]: no ES6

File: js/libs/builder.js [hunk 5]
```diff
@@ -191,11 +196,117 @@ var app = new Vue({
...
     save: function() {
       return Fliplet.Widget.save(this.settings);
+    },
+    configureEmailTemplate: function() {
+      var $vm = this;
+      var allFields = $vm.settings.fields;
+      // Creates default email template
+      var defaultEmailTemplate = '<h1>' + $vm.settings.displayName + '</h1><p>A new form submission has been received.</p>';
+      defaultEmailTemplate += '<ul>';
+      for (var i = 0; i < allFields.length; i++) {
```
Comment (squallstar) [inline js/libs/builder.js]: can't you just use `$vm.settings.fields.forEach` and avoid using a `for` cycle and `allFields[i]` multiple times?

File: js/libs/builder.js [hunk 6]
```diff
@@ -191,11 +196,117 @@ var app = new Vue({
...
+                hooks: dataSource.hooks
+              });
+            }
+          });
+        }
+
+        Fliplet.Widget.autosize();
+      });
+    },
+    configureGenerateEmailDevice: function() {
```
Comment (squallstar) [inline js/libs/builder.js]: configure or generate?

File: js/libs/builder.js [hunk 7]
```diff
@@ -191,11 +196,117 @@ var app = new Vue({
...
+      var $vm = this;
+      var allFields = $vm.settings.fields;
+      // Creates default email template
+      var defaultEmailTemplate = '<h1>' + $vm.settings.displayName + '</h1><p>A form submission has been received.</p>';
+      defaultEmailTemplate += '<ul>'
+      for (var i = 0; i < allFields.length; i++) {
```
Comment (squallstar) [inline js/libs/builder.js]: pretty sure I've seen this already above. Why is this whole code duplicated?

File: js/libs/builder.js [hunk 8]
```diff
@@ -204,8 +315,8 @@ var app = new Vue({
       if (newVal) {
         Fliplet.Studio.emit('widget-mode', 'wide');
         setTimeout(() => {
-          attatchObservers()
-        }, 1)
+          attatchObservers();
```
Comment (squallstar) [inline js/libs/builder.js]: typo in attach

File: js/libs/builder.js [hunk 9]
```diff
@@ -204,8 +315,8 @@ var app = new Vue({
       if (newVal) {
         Fliplet.Studio.emit('widget-mode', 'wide');
         setTimeout(() => {
```
Comment (squallstar) [inline js/libs/builder.js]: no ES6

File: js/libs/builder.js [hunk 10]
```diff
@@ -234,22 +345,23 @@ var app = new Vue({
       this.settings = generateFormDefaults(settings);
       this.fields = this.settings.fields;
 
-      this.save().then(function() {
+      this.save().then(() => {
```
Comment (squallstar) [inline js/libs/builder.js]: no ES6

File: js/libs/builder.js [hunk 11]
```diff
@@ -234,22 +345,23 @@ var app = new Vue({
...
       });
     },
     'section': function(value) {
+      var $vm = this;
       if (value === 'settings') {
-        changeSelectText()
+        changeSelectText();
 
         if (!this.resultEditor) {
           setTimeout(() => {
```
Comment (squallstar) [inline js/libs/builder.js]: no ES6

File: js/libs/builder.js [hunk 12]
```diff
@@ -258,11 +370,51 @@ var app = new Vue({
...
-          setTimeout(function() {
+          setTimeout(() => {
+            $vm.resultEditor.refresh();
+          }, 1);
+        }
+      }
+    },
+    'settings.onSubmit': function(array) {
+      var $vm = this;
+      this.showDataSource = array.indexOf("dataSource") > -1 ? true : false;
```
Comment (squallstar) [inline js/libs/builder.js]: no need for `?true:false`

File: js/libs/builder.js [hunk 13]
```diff
@@ -353,17 +515,21 @@ var app = new Vue({
...
+            $vm.setupCodeEditor();
+          }, 1);
         }
 
         return;
       }
 
       // Save and close
-      $vm.save().then(function() {
+      $vm.save().then(() => {
```
Comment (squallstar) [inline js/libs/builder.js]: no ES6

File: js/libs/builder.js [hunk 14]
```diff
@@ -353,17 +515,21 @@ var app = new Vue({
...
         Fliplet.Widget.complete();
       });
     }
+
+    Fliplet.API.request('v1/user').then(function(response) {
+      window.userData = response.user;
```
Comment (squallstar) [inline js/libs/builder.js]: why set it to the window? can you keep it private?

Comment (hcarneiro) [inline js/libs/builder.js]: @squallstar I need to use that data later in the code, how can I set the var on a global level to the entire `builder.js` code but keep it private?

File: js/libs/builder.js (hunk 1 above)
Comment (squallstar) [inline js/libs/builder.js]: @tonytlwu because we don't compile these files with babel, hence using ES6 might fail where it's not supported to I asked to revert to "ol' plain ES5".
